### PR TITLE
document for ACR-1428

### PR DESCRIPTION
**Pull request by haejeongkim**
Add new APIs for ability

### CODE REVIEW

File: docs/application/native/guides/multimedia/media-controller.md [hunk 1]
```diff
@@ -1023,6 +1105,26 @@ The following table lists all the search category attributes the server can rece
...
+| `MC_ABILITY_SHUFFLE`             | Ability for shuffle                      |
+| `MC_ABILITY_REPEAT`              | Ability for repeat                       |
+| `MC_ABILITY_PLAYBACK_POSITION`   | Ability for playback position            |
+| `MC_ABILITY_PLAYLIST`            | Ability for playlist                     |
+| `MC_ABILITY_CLIENT_CUSTOM`       | Ability for custom event                 |
+| `MC_ABILITY_SEARCH`              | Ability for search                       |
+
+> **Note**
+>
+> This Attributes support Tizen 5.5 and Higher for Mobile and Wearable.
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/media-controller.md]: > These attributes support Tizen 5.5 and Higher for Mobile and Wearable.

File: docs/application/native/guides/multimedia/media-controller.md [hunk 2]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
 > This feature supports Tizen 5.0 and Higher for Mobile and Wearable.
 
 
+## Updating and Retrieving Abilities
+To update the abilities on the server side:
+
+1. Create the media controller server handle using the `mc_server_create()`:
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/media-controller.md]: 1. Create the media controller server handle using `mc_server_create()`:

File: docs/application/native/guides/multimedia/media-controller.md [hunk 3]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+   ret = mc_server_set_display_mode_ability(g_mc_server, MC_DISPLAY_MODE_LETTER_BOX | MC_DISPLAY_MODE_ORIGIN_SIZE | MC_DISPLAY_MODE_FULL_SCREEN | MC_DISPLAY_MODE_CROPPED_FULL, MC_ABILITY_SUPPORTED_YES);
+   ```
+   
+   For example, to update the display rotation ability, set the ability using the 'mc_server_set_display_rotation_ability()':
+    ```
+   ret = mc_server_set_display_rotation_ability(g_mc_server, MC_DISPLAY_ROTATION_NONE | MC_DISPLAY_ROTATION_90 | MC_DISPLAY_ROTATION_180 | MC_DISPLAY_ROTATION_270, MC_ABILITY_SUPPORTED_YES);
+   ```
+   
+   For other abilities, to update the ability, set the ability using the 'mc_server_set_ability_support()':
+   For example, to update shuffle and repeat ability, set the ability using the 'mc_server_set_ability_support()':
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/media-controller.md]: To update shuffle and repeat ability, set the ability using 'mc_server_set_ability_support()':

File: docs/application/native/guides/multimedia/media-controller.md [hunk 4]
```diff
@@ -71,6 +71,18 @@ the applications you want.
   >
   > This feature supports Tizen 5.0 and Higher for Mobile and Wearable.
 
+- Updating and retrieving abilities
+
+  You can [update the abilities](#updating-and-retrieving-abilities) on the server side, and then retrieve the abilities on the client side.
+
+  The media controller server provides current abilities about the registered application.
+  
+  Only When the server support abilities, the media controller clients can send command to server.
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: When the server supports abilities, then the media controller clients can send commands to the server.
=> I updated like that.

File: docs/application/native/guides/multimedia/media-controller.md [hunk 5]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
 > This feature supports Tizen 5.0 and Higher for Mobile and Wearable.
 
 
+## Updating and Retrieving Abilities
+To update the abilities on the server side:
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md [hunk 6]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+
+1. Create the media controller server handle using the `mc_server_create()`:
+
+   ```
+   ret = mc_server_create(&g_server_h);
+   ```
+   
+2. Set the abilities using the corresponding 'mc_server_set_XXX_ability()', or 'mc_server_set_ability_support()':
+   For example, to update the playback ability, set the ability for each playback actions using the 'mc_server_set_playback_ability()', and then update the abilities using the 'mc_server_update_playback_ability()':
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: I updated. 
using 'mc_server_update_playback_ability()' is correct? or 
using the 'mc_server_update_playback_ability()' is correct?

File: docs/application/native/guides/multimedia/media-controller.md [hunk 7]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+   
+   ret = mc_server_update_playback_ability(g_mc_server);
+   ```
+   
+   For example, to update the display mode ability, set the ability using the 'mc_server_set_display_mode_ability()':
+   ```
+   ret = mc_server_set_display_mode_ability(g_mc_server, MC_DISPLAY_MODE_LETTER_BOX | MC_DISPLAY_MODE_ORIGIN_SIZE | MC_DISPLAY_MODE_FULL_SCREEN | MC_DISPLAY_MODE_CROPPED_FULL, MC_ABILITY_SUPPORTED_YES);
+   ```
+   
+   For example, to update the display rotation ability, set the ability using the 'mc_server_set_display_rotation_ability()':
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md [hunk 8]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+    ```
+   ret = mc_server_set_display_rotation_ability(g_mc_server, MC_DISPLAY_ROTATION_NONE | MC_DISPLAY_ROTATION_90 | MC_DISPLAY_ROTATION_180 | MC_DISPLAY_ROTATION_270, MC_ABILITY_SUPPORTED_YES);
+   ```
+   
+   For other abilities, to update the ability, set the ability using the 'mc_server_set_ability_support()':
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Option 2. Done

File: docs/application/native/guides/multimedia/media-controller.md [hunk 9]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+   For example, to update shuffle and repeat ability, set the ability using the 'mc_server_set_ability_support()':
+   ```
+   ret = mc_server_set_ability_support(g_mc_server, MC_ABILITY_SHUFFLE, MC_ABILITY_SUPPORTED_YES);
+   ret = mc_server_set_ability_support(g_mc_server, MC_ABILITY_REPEAT, MC_ABILITY_SUPPORTED_NO);
+   ```
+
+To retrieve the abilities on the client side:
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md [hunk 10]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+
+1. Create the media controller client handle using the `mc_client_create()`:
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: remove "the"?

File: docs/application/native/guides/multimedia/media-controller.md [hunk 11]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+
+   ```
+   ret = mc_client_create(&g_client_h);
+   ```
+
+2. Define the callback that is invoked when the client receives the abilities changes.
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md [hunk 12]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+   - To register a callback for playback ability change, use the `mc_client_set_playback_ability_updated_cb()`.
+   - To register a callback for display mode ability change, use the `mc_client_set_display_mode_ability_updated_cb()`.
+   - To register a callback for display rotation ability change, use the `mc_client_set_display_rotation_ability_updated_cb()`.
+   - To register a callback for other abilities change, use the `mc_client_set_ability_support_updated_cb()`.
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Can you check below pr regarding this?
https://github.com/Samsung/tizen-docs/pull/881

File: docs/application/native/guides/multimedia/media-controller.md [hunk 13]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+   
+   For example, to define a callback for a playback ability:
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md [hunk 14]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+   ```
+   ret = mc_client_set_playback_ability_updated_cb(g_client_h, _mc_playback_ability_updated_cb, NULL);
+   ```
+
+3. Destroy the media controller client handle using the `mc_client_destroy()`, when media controller client handle is no longer needed:
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Remove "the".
But as I remember, I add "the". because you requested it to me.

File: docs/application/native/guides/multimedia/media-controller.md [hunk 15]
```diff
@@ -1023,6 +1105,26 @@ The following table lists all the search category attributes the server can rece
 >
 > These Attributes support Tizen 5.0 and Higher for Mobile and Wearable.
 
+## Media Controller Search Category Attributes
+
+The following table lists all the search category attributes the server can receive:
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md [hunk 16]
```diff
@@ -1023,6 +1105,26 @@ The following table lists all the search category attributes the server can rece
...
+
+**Table: Media controller ability attributes**
+
+| Attribute                        | Description                              |
+|----------------------------------|------------------------------------------|
+| **Ability**                      |                                          |
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: I just consisted with other attribute tables.
Add "ability"? What's your opinion?

File: docs/application/native/guides/multimedia/media-controller.md (hunk 3 above)
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: shuffle and repeat is one of the example for "mc_server_set_ability_support()".
Is it OK to remove "for example"?

File: docs/application/native/guides/multimedia/media-controller.md [hunk 17]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
 > This feature supports Tizen 5.0 and Higher for Mobile and Wearable.
 
 
+## Updating and Retrieving Abilities
+To update the abilities on the server side:
+
+1. Create the media controller server handle using the `mc_server_create()`:
+
+   ```
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Could you explain more detail? what's wrong?

File: docs/application/native/guides/multimedia/media-controller.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (nawab-samsung) [reply docs/application/native/guides/multimedia/media-controller.md]: using 'mc_server_update_playback_ability()' is correct 

why I removed "the" is because there exists no qualifier after the callback name for example 

Just for your understanding: 

using the 'xxx_yyy_zzz()` callback. 

Here we have the word "callback" as the qualifier, thus we write "the" before the callback name.

File: docs/application/native/guides/multimedia/media-controller.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (nawab-samsung) [reply docs/application/native/guides/multimedia/media-controller.md]: yes @haejeongkim

File: docs/application/native/guides/multimedia/media-controller.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (nawab-samsung) [reply docs/application/native/guides/multimedia/media-controller.md]: @haejeongkim 
Please see the given clarification about removing "the", request you to remove "the"

File: docs/application/native/guides/multimedia/media-controller.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (nawab-samsung) [reply docs/application/native/guides/multimedia/media-controller.md]: From the left side, code must be indented to 4 spaces. 


    ````
    code starts here`
    {
     xyz();
    }
    ````

File: docs/application/native/guides/multimedia/media-controller.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (nawab-samsung) [reply docs/application/native/guides/multimedia/media-controller.md]: 940: 
-  `mc_client_set_playback_ability_updated_cb()`: To register a callback for changing the playback ability.
941: 
-  `mc_client_set_display_mode_ability_updated_cb()`: To register a callback for changing the display mode ability.
942:
 - `mc_client_set_display_rotation_ability_updated_cb()`: To register a callback for changing the display rotation ability.
943: 
- `mc_client_set_ability_support_updated_cb()`: To register a callback for changing other abilities. 

Similarly, then we need to change other parts of the file also to maintain consistency: 




@haejeongkim 
Thanks 

As I see, @mijin85cho suggested the following way, let us align to that and change the following lines accordingly. 
Line 613 -618 
613:
- `mc_server_set_playback_action_cmd_received_cb()`: To register a callback for playback state change commands.
614:
- `mc_server_set_playback_position_cmd_received_cb()`: To register a callback for playback position change commands. 
615:
- mc_server_set_shuffle_mode_cmd_received_cb()`: To register a callback for shuffle mode change commands, use `
616:
- `mc_server_set_repeat_mode_cmd_received_cb()`: To register a callback for repeat mode change commands.
617:
- `mc_server_set_playlist_cmd_received_cb()`: To register a callback for played item, playback state, and playback position change commands in playlist. 
618:
-`mc_server_set_custom_cmd_received_cb()`.To register a callback for a custom command, use

File: docs/application/native/guides/multimedia/media-controller.md (hunk 3 above)
Comment (nawab-samsung) [reply docs/application/native/guides/multimedia/media-controller.md]: Yes @haejeongkim 
Global: Suggestion: How about removing "for example":
If you agree, we need to add the following line after line after 924.

The following example demonstrates setting up various abilities:

File: docs/application/native/guides/multimedia/media-controller.md [hunk 18]
```diff
@@ -875,6 +887,89 @@ To process the received search command on the server side, follow these steps:
...
+   mc_playback_ability_h ability = NULL;
+   ret = mc_client_get_server_playback_ability(g_client_h, "server_name", &ability);
+   ```
+
+   - `mc_client_get_server_display_mode_ability()`: For display mode ability.
+   - `mc_client_get_server_display_rotation_ability()`: For display rotation ability.
+   - `mc_client_get_server_ability_support()`: For other abilities.
+   
+4. Destroy the media controller client handle using the `mc_client_destroy()`, when media controller client handle is no longer needed:
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/media-controller.md]: 4. Destroy the media controller client handle using `mc_client_destroy()`,  when media controller client handle is no longer needed:

File: docs/application/native/guides/multimedia/media-controller.md [hunk 19]
```diff
@@ -875,6 +887,89 @@ To process the received search command on the server side, follow these steps:
...
+1. Create the media controller server handle using the `mc_server_create()`:
+
+   ```
+   ret = mc_server_create(&g_server_h);
+   ```
+   
+2. Set the abilities using the corresponding 'mc_server_set_XXX_ability()', or 'mc_server_set_ability_support()'.
+   The following examples demonstrate setting up various abilities:
+   
+   To update the playback ability, set the ability for each playback action using the 'mc_server_set_playback_ability()', and then update the ability using the 'mc_server_update_playback_ability()':
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/media-controller.md]: To update the playback ability, set the ability for each playback action using 'mc_server_set_playback_ability()', and then update the ability using  'mc_server_update_playback_ability()':

Remove "the"

File: docs/application/native/guides/multimedia/media-controller.md [hunk 20]
```diff
@@ -875,6 +887,89 @@ To process the received search command on the server side, follow these steps:
...
+   ret = mc_client_set_playback_ability_updated_cb(g_client_h, _mc_playback_ability_updated_cb, NULL);
+   ```
+   
+   - `mc_client_set_playback_ability_updated_cb()`: For playback ability change.
+   - `mc_client_set_display_mode_ability_updated_cb()`: For display mode ability change.
+   - `mc_client_set_display_rotation_ability_updated_cb()`: For display rotation ability change.
+   - `mc_client_set_ability_support_updated_cb()`: For other abilities change.
+   
+3. Or, client can get the server's ability directly.
+   For example, to get a server's playback ability:
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/media-controller.md]: To get server playback ability:

File: docs/application/native/guides/multimedia/media-controller.md [hunk 21]
```diff
@@ -875,6 +887,89 @@ To process the received search command on the server side, follow these steps:
...
+   ```
+   ret = mc_client_set_playback_ability_updated_cb(g_client_h, _mc_playback_ability_updated_cb, NULL);
+   ```
+   
+   - `mc_client_set_playback_ability_updated_cb()`: For playback ability change.
+   - `mc_client_set_display_mode_ability_updated_cb()`: For display mode ability change.
+   - `mc_client_set_display_rotation_ability_updated_cb()`: For display rotation ability change.
+   - `mc_client_set_ability_support_updated_cb()`: For other abilities change.
+   
+3. Or, client can get the server's ability directly.
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/media-controller.md]: The client can get the server ability directly.
The following example demonstrates setting up the server abilities:

File: docs/application/native/guides/multimedia/media-controller.md [hunk 22]
```diff
@@ -1023,6 +1105,26 @@ The following table lists all the search category attributes the server can rece
 >
 > These Attributes support Tizen 5.0 and Higher for Mobile and Wearable.
 
+## Media Controller Search Category Attributes
```
Comment (mijin85cho) [inline docs/application/native/guides/multimedia/media-controller.md]: Would you please check this title? Is this table really related with Search?

File: docs/application/native/guides/multimedia/media-controller.md [hunk 23]
```diff
@@ -875,6 +887,89 @@ To process the received search command on the server side, follow these steps:
...
+To update the abilities on the server side, follow these steps:
+
+1. Create the media controller server handle using the `mc_server_create()`:
+
+   ```
+   ret = mc_server_create(&g_server_h);
+   ```
+   
+2. Set the abilities using the corresponding 'mc_server_set_XXX_ability()', or 'mc_server_set_ability_support()'.
+   The following examples demonstrate setting up various abilities:
```
Comment (mijin85cho) [inline docs/application/native/guides/multimedia/media-controller.md]: Please change line from 899:

Set the abilities using the corresponding `mc_server_set_XXX_ability()`, or `mc_server_set_ability_support()`.


Also, @nawab-samsung, I'd like to delete "The following examples demonstrate setting up various abilities:" on line 900 and keep current "To update ~".
I don't think this layout is good.
image

If we'd like to add "The following examples demonstrate setting up various abilities:" , the layout should be changed a little bit more as like : 

---

Set the abilities using the corresponding 'mc_server_set_XXX_ability()', or 'mc_server_set_ability_support()'.

 The following examples demonstrate setting up various abilities:

   - To update the playback ability,~

   - To update the display mode ability, 
---

**I don't want to do that.**. So please remove added line 900.

File: docs/application/native/guides/multimedia/media-controller.md (hunk 19 above)
Comment (mijin85cho) [reply docs/application/native/guides/multimedia/media-controller.md]: Please change ' to `.

 To update the playback ability, set the ability for each playback action using the `mc_server_set_playback_ability()`, and then update the ability using the `mc_server_update_playback_ability()`:

File: docs/application/native/guides/multimedia/media-controller.md [hunk 24]
```diff
@@ -875,6 +887,89 @@ To process the received search command on the server side, follow these steps:
...
+To retrieve the abilities on the client side, follow these steps:
+
+1. Create the media controller client handle using the `mc_client_create()`:
+
+   ```
+   ret = mc_client_create(&g_client_h);
+   ```
+
+2. Define the callback that gets invoked when the client receives the change in abilities.
+   To define a callback for a playback ability:
```
Comment (mijin85cho) [inline docs/application/native/guides/multimedia/media-controller.md]: As the context flow is a little bit changed, how about inserting a line between 941 and 942 to separate them?

File: docs/application/native/guides/multimedia/media-controller.md (hunk 21 above)
Comment (mijin85cho) [reply docs/application/native/guides/multimedia/media-controller.md]: I think the suggestion is too wordy. 

The client can get the server ability directly. For example to get server playback ability:

Would be enough.

File: docs/application/native/guides/multimedia/media-controller.md (hunk 20 above)
Comment (mijin85cho) [reply docs/application/native/guides/multimedia/media-controller.md]: Please refer my comment on line 952.

File: docs/application/native/guides/multimedia/media-controller.md [hunk 25]
```diff
@@ -875,6 +887,76 @@ To process the received search command on the server side, follow these steps:
...
+## Updating and Retrieving Abilities
+To update the abilities on the server side:
+
+1. Create the media controller server handle using the `mc_server_create()`:
+
+   ```
+   ret = mc_server_create(&g_server_h);
+   ```
+   
+2. Set the abilities using the corresponding 'mc_server_set_XXX_ability()', or 'mc_server_set_ability_support()':
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Ok.

File: docs/application/native/guides/multimedia/media-controller.md (hunk 2 above)
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 18 above)
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 19 above)
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 20 above)
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 21 above)
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md [hunk 26]
```diff
@@ -71,6 +71,18 @@ the applications you want.
   >
   > This feature supports Tizen 5.0 and Higher for Mobile and Wearable.
 
+- Updating and retrieving abilities
+
+  You can [update the abilities](#updating-and-retrieving-abilities) on the server side, and then retrieve the abilities on the client side.
+
+  The media controller server provides the update and retrieval abilities to the registered application.
```
[the remark this answers is not in the document]
Comment (nawab-samsung) [reply docs/application/native/guides/multimedia/media-controller.md]: @mijin85cho thanks,  
1. I do agree with you.  
2. Why I added  "update and retrieval" is because, the whole section is talking about two abilities, which have multiple associated callbacks. 

@haejeongkim as requested by @mijin85cho , can you please provide your valuable opinion on this.

[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: The media controller server provides current abilities about the registered application.
=> I updated like this.

File: docs/application/native/guides/multimedia/media-controller.md [hunk 27]
```diff
@@ -875,6 +887,89 @@ To process the received search command on the server side, follow these steps:
...
+   ret = mc_server_set_playback_ability(g_mc_server, MC_PLAYBACK_ACTION_NEXT, MC_ABILITY_SUPPORTED_NO);
+   ret = mc_server_set_playback_ability(g_mc_server, MC_PLAYBACK_ACTION_PREV, MC_ABILITY_SUPPORTED_YES);
+   ret = mc_server_set_playback_ability(g_mc_server, MC_PLAYBACK_ACTION_FAST_FORWARD, MC_ABILITY_SUPPORTED_NO);
+   ret = mc_server_set_playback_ability(g_mc_server, MC_PLAYBACK_ACTION_REWIND, MC_ABILITY_SUPPORTED_YES);
+   ret = mc_server_set_playback_ability(g_mc_server, MC_PLAYBACK_ACTION_TOGGLE_PLAY_PAUSE, MC_ABILITY_SUPPORTED_NO);
+   
+   ret = mc_server_update_playback_ability(g_mc_server);
+   ```
+   
+   To update the display mode ability, set the ability using 'mc_server_set_display_mode_ability()':
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 23 above)
Comment (nawab-samsung) [reply docs/application/native/guides/multimedia/media-controller.md]: @mijin85cho 
1. Just added the line to get continuity and the flow. Also to make things more lucid. Moreover, I wished to keep the context of "For example" intact. 

@haejeongkim , Request you to remove line 900

File: docs/application/native/guides/multimedia/media-controller.md [hunk 28]
```diff
@@ -875,6 +887,89 @@ To process the received search command on the server side, follow these steps:
...
+   ```
+   ret = mc_server_set_display_mode_ability(g_mc_server, MC_DISPLAY_MODE_LETTER_BOX | MC_DISPLAY_MODE_ORIGIN_SIZE | MC_DISPLAY_MODE_FULL_SCREEN | MC_DISPLAY_MODE_CROPPED_FULL, MC_ABILITY_SUPPORTED_YES);
+   ```
+   
+   To update the display rotation ability, set the ability using 'mc_server_set_display_rotation_ability()':
+    ```
+   ret = mc_server_set_display_rotation_ability(g_mc_server, MC_DISPLAY_ROTATION_NONE | MC_DISPLAY_ROTATION_90 | MC_DISPLAY_ROTATION_180 | MC_DISPLAY_ROTATION_270, MC_ABILITY_SUPPORTED_YES);
+   ```
+   
+   In case of other abilities, set the ability using 'mc_server_set_ability_support()', to update an ability:
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md [hunk 29]
```diff
@@ -875,6 +887,89 @@ To process the received search command on the server side, follow these steps:
...
+   
+   ret = mc_server_update_playback_ability(g_mc_server);
+   ```
+   
+   To update the display mode ability, set the ability using 'mc_server_set_display_mode_ability()':
+   ```
+   ret = mc_server_set_display_mode_ability(g_mc_server, MC_DISPLAY_MODE_LETTER_BOX | MC_DISPLAY_MODE_ORIGIN_SIZE | MC_DISPLAY_MODE_FULL_SCREEN | MC_DISPLAY_MODE_CROPPED_FULL, MC_ABILITY_SUPPORTED_YES);
+   ```
+   
+   To update the display rotation ability, set the ability using 'mc_server_set_display_rotation_ability()':
```
[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 22 above)
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md (hunk 23 above)
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: I removed "The following examples demonstrate setting up various abilities:"

File: docs/application/native/guides/multimedia/media-controller.md (hunk 24 above)
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

File: docs/application/native/guides/multimedia/media-controller.md [hunk 30]
```diff
@@ -1021,7 +1126,63 @@ The following table lists all the search category attributes the server can rece
...
+| `MC_ABILITY_SHUFFLE`             | Ability for shuffle                      |
+| `MC_ABILITY_REPEAT`              | Ability for repeat                       |
+| `MC_ABILITY_PLAYBACK_POSITION`   | Ability for playback position            |
+| `MC_ABILITY_PLAYLIST`            | Ability for playlist                     |
+| `MC_ABILITY_CLIENT_CUSTOM`       | Ability for custom event                 |
+| `MC_ABILITY_SEARCH`              | Ability for search                       |
+
+> **Note**
+>
+> These Attributes support Tizen 5.5 and Higher for Mobile and Wearable.
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/media-controller.md]: My bad, I missed it. 
Much thanks to @mijin85cho for pointing it out. 
@haejeongkim 
Kindly change Attributes to attributes. 
> These attributes support Tizen 5.5 and Higher for Mobile and Wearable.

[the remark this answers is not in the document]
Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done

Comment (haejeongkim) [reply docs/application/native/guides/multimedia/media-controller.md]: Done
